pontoon/teams/views.py

```diff
@@ -379,6 +384,90 @@ def ajax_translation_memory_delete(request, locale):
     return HttpResponse("ok")
 
 
+@require_AJAX
+@require_POST
+@permission_required_or_403("base.can_translate_locale", (Locale, "code", "locale"))
+@transaction.atomic
+def ajax_translation_memory_upload(request, locale):
+    """Upload Translation Memory entries from a .TMX file."""
+    try:
+        file = request.FILES["tmx_file"]
+    except MultiValueDictKeyError:
+        return JsonResponse(
+            {"status": False, "message": "No file uploaded."},
+            status=400,
+        )
+
+    if not file.name.endswith(".tmx"):
+        return JsonResponse(
+            {
+                "status": False,
+                "message": "Invalid file format. Only .TMX files are supported.",
+            },
+            status=400,
+        )
+
+    locale = get_object_or_404(Locale, code=locale)
+    code = locale.code
+
+    # Parse the TMX file
+    try:
+        tree = ET.parse(file)
+        root = tree.getroot()
+    except ET.ParseError as e:
+        return JsonResponse(
+            {"status": False, "message": f"Invalid XML file: {e}"}, status=400
+        )
+
+    # Extract TM entries
+    file_entries = []
+    ns = {"xml": "http://www.w3.org/XML/1998/namespace"}
+    for tu in root.findall(".//tu"):
+        try:
+            source = tu.find("./tuv[@xml:lang='en-US']/seg", namespaces=ns).text.strip()
+            target = tu.find(
+                f"./tuv[@xml:lang='{code}']/seg", namespaces=ns
+            ).text.strip()
+
+            if source and target:
+                file_entries.append({"source": source, "target": target})
+        except Exception as e:
+            log.info(f"Error processing <tu>: {e}")
```

Comment:
@eemeli Do you plan to add support for TMX parsing / serializing to moz.l10n? Until this is clear, I just baked the parser in the view, because it's fairly simple.

Reply:
Also, should we be less strict about the TMX format?

Reply:
No current plans for TMX support in moz.l10n. The format is sufficiently different in structure that it doesn't really fit into the message & resource data models, and so would need a separate, different representation.